File: Armstrong.py
def is_armstrong(n):
    if n < 0:
        return False
    temp = n
    digits = 0
    while temp != 0:
        digits += 1
        temp //= 10
    temp = n
    total = 0
    while temp != 0:
        digit = temp % 10
        power = 1
        for _ in range(digits):
            power *= digit
        total += power
        temp //= 10
    return total == n

num = int(input("Enter a non-negative integer: "))
if is_armstrong(num):
    print(f"{num} is an Armstrong number.")
else:
    print(f"{num} is not an Armstrong number.")

File: Armstrong.java
import java.util.Scanner;

public class Armstrong {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter a non-negative integer: ");
        int num = input.nextInt();
        int original = num;
        int sum = 0;
        int digits = 0;
        int temp = num;
        while (temp != 0) {
            digits++;
            temp /= 10;
        }
        temp = num;
        while (temp != 0) {
            int digit = temp % 10;
            int power = 1;
            for (int i = 0; i < digits; i++) {
                power *= digit;
            }
            sum += power;
            temp /= 10;
        }
        if (sum == original) {
            System.out.println(original + " is an Armstrong number.");
        } else {
            System.out.println(original + " is not an Armstrong number.");
        }
    }
}